packages/mui-base/src/useNumberField/useNumberField.ts

```diff
@@ -0,0 +1,802 @@
+    unsubscribeFromGlobalContextMenuRef.current();
+    movesAfterTouchRef.current = 0;
+  });
+
+  const changeValue = useEventCallback((amount: number, dir: 1 | -1) => {
+    const rawValue = rawValueRef.current;
+    const nextValue =
+      typeof rawValue === 'number' ? rawValue + amount * dir : Math.max(0, min ?? 0);
+    rawValueRef.current = nextValue;
+    setValue(nextValue);
+  });
+
+  const increment = useEventCallback((amount: number) => {
+    changeValue(amount, 1);
+  });
+
+  const decrement = useEventCallback((amount: number) => {
+    changeValue(amount, -1);
+  });
+
+  const startAutoChange = useEventCallback((isIncrement: boolean) => {
+    stopAutoChange();
+
+    if (!inputRef.current) {
+      return;
+    }
+
+    const win = ownerWindow(inputRef.current);
+
+    // A global context menu is necessary to prevent the context menu from appearing when the touch
+    // is slightly outside of the element's hit area.
+    win.addEventListener('contextmenu', handleGlobalContextMenu);
+    unsubscribeFromGlobalContextMenuRef.current = () => {
+      win.removeEventListener('contextmenu', handleGlobalContextMenu);
+    };
+
+    win.addEventListener(
+      'pointerup',
+      () => {
+        isPressedRef.current = false;
+        stopAutoChange();
+      },
+      { once: true },
+    );
+
+    function tick() {
+      const amount = getStepAmount();
+      if (isIncrement) {
+        increment(amount);
+      } else {
+        decrement(amount);
+      }
+    }
+
+    tick();
+
+    startTickTimeoutRef.current = window.setTimeout(() => {
+      tickIntervalRef.current = window.setInterval(tick, CHANGE_VALUE_TICK_DELAY);
+    }, START_AUTO_CHANGE_DELAY);
+  });
+
+  // We need to update the input value when the external `value` prop changes. This ends up acting
+  // as a single source of truthto update the input value, bypassing the need to manually set it in
+  // each event handler internally in this hook.
+  // This is done inside a layout effect as an alternative to the technique to set state during
+  // render as we're accessing a ref, which must be inside an effect.
+  // https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
+  //
+  // ESLint is disabled because it needs to run even if the parsed value hasn't changed, since the
+  // value still can be formatted differently.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEnhancedEffect(function syncFormattedInputValueOnValueChange() {
+    // This ensures the value is only updated on blur rather than every keystroke, but still
+    // allows the input value to be updated when the value is changed externally.
+    if (!allowInputSyncRef.current) {
+      return;
+    }
+
+    const nextInputValue = formatNumber(value, formatOptionsRef.current);
+
+    if (nextInputValue !== inputValue) {
+      setInputValue(nextInputValue);
+    }
+  });
+
+  React.useEffect(
+    function stopAutoChangeOnUnmount() {
```

Comment:
Do we need to name this function?

Reply:
It's small and clear, so probably not. I just added it for consistency since I named every effect, but I can remove this one.